FastLSTM.lua

```diff
@@ -91,8 +115,8 @@ function FastLSTM:nngraphModel()
    local x, prev_h, prev_c = unpack(inputs)
    
    -- evaluate the input sums at once for efficiency
-   local i2h = self.i2g(x):annotate{name='i2h'}
-   local h2h = self.o2g(prev_h):annotate{name='h2h'}
+   local i2h = bn_wx(self.i2g(x):annotate{name='i2h'}):annotate {name='bn_wx'}
```

Comment:
We are also batch normalizing the `next_c` after these operations and between batch normalizing `i2h` , `h2h` and `next_c`, we have to do some computations for the gates in-between. If I go with your direction, this would make the code longer. The `nn.Identty()` takes care of the redundancy. Okay anyway if you prefer it to be that way. 

Your implementation would look like 

```lua
   local bn_wx, bn_wh, bn_c  
   if self.bn then  
      bn_wx = nn.BatchNormalization(4*self.outputSize, self.eps, self.momentum, self.affine)
      bn_wh = nn.BatchNormalization(4*self.outputSize, self.eps, self.momentum, self.affine)
      bn_c  = nn.BatchNormalization(self.outputSize, self.eps, self.momentum, self.affine)

      
      -- evaluate the input sums at once for efficiency
      local i2h = bn_wx(self.i2g(x):annotate{name='i2h'}):annotate {name='bn_wx'}
      local h2h = bn_wh(self.o2g(prev_h):annotate{name='h2h'}):annotate {name = 'bn_wh'}
      local all_input_sums = nn.CAddTable()({i2h, h2h})

      local reshaped = nn.Reshape(4, self.outputSize)(all_input_sums)
      -- input, hidden, forget, output
      local n1, n2, n3, n4 = nn.SplitTable(2)(reshaped):split(4)
      local in_gate = nn.Sigmoid()(n1)
      local in_transform = nn.Tanh()(n2)
      local forget_gate = nn.Sigmoid()(n3)
      local out_gate = nn.Sigmoid()(n4)
      
      -- perform the LSTM update
      local next_c           = nn.CAddTable()({
        nn.CMulTable()({forget_gate, prev_c}),
        nn.CMulTable()({in_gate,     in_transform})
      })
      -- gated cells form the output
      local next_h = nn.CMulTable()({out_gate, nn.Tanh()(bn_c(next_c):annotate {name = 'bn_c'}) })
   else
      -- evaluate the input sums at once for efficiency
      local i2h = self.i2g(x):annotate{name='i2h'}
      local h2h = self.o2g(prev_h):annotate{name='h2h'}
      local all_input_sums = nn.CAddTable()({i2h, h2h})

      local reshaped = nn.Reshape(4, self.outputSize)(all_input_sums)
      -- input, hidden, forget, output
      local n1, n2, n3, n4 = nn.SplitTable(2)(reshaped):split(4)
      local in_gate = nn.Sigmoid()(n1)
      local in_transform = nn.Tanh()(n2)
      local forget_gate = nn.Sigmoid()(n3)
      local out_gate = nn.Sigmoid()(n4)
      
      -- perform the LSTM update
      local next_c           = nn.CAddTable()({
        nn.CMulTable()({forget_gate, prev_c}),
        nn.CMulTable()({in_gate,     in_transform})
      })
      -- gated cells form the output
      local next_h = nn.CMulTable()({out_gate, nn.Tanh()(next_c)})
   end
```

You want me to send the PR as suggested?